plaidplotter.py

```diff
@@ -0,0 +1,318 @@
+
+        fig, axes = plot(df, "model", col_order, max_time)
+        labels = (list(set(df['model'])))
+        set_labels(fig, axes, labels, list(set(df['batch'])), len(labels))
+        color_bars(axes, palette, len(df['model']), len(list(set(df['batch']))))
+
+        # saving graph
+        title = ''
+        if title_str != '':
+            title = title_str + '.png'
+        else:
+            title = time.strftime("plaidbench %Y-%m-%d-%H:%M.png")
+        print("saving figure at '" + path_str + '/' + title + "'\n")
+        fig.savefig(path_str + '/' + title)
+
+        # importing golden npy files, not other golden file utilization though
+        if isProspector == True:
+            models = list(set(df['model']))
+            batches = list(set(df['batch']))
+
+            this_dir = os.path.dirname(os.path.abspath(__file__))
+            golden_path = os.path.join(this_dir, 'golden')
+
+            illusory_path = ''
+            GOLD = '\033[0;33m'
+            BGOLD = '\033[1;33m'
+            PURPLE = '\033[0;35m'
+            ENDC = '\033[0m'
+
+            print('Attempting to retrieve ' + BGOLD + 'Golden Files' + ENDC + '...\n')
+            for model in models:
+                illusory_path = os.path.join(golden_path, model)
+                for batch in batches:
+                    if isTrain == True:
+                        filename = '{},bs-{}.npy'.format('train', batch)
+                    else:
+                        filename = '{},bs-{}.npy'.format('infer', batch)
+                    illusory_path = os.path.join(golden_path, model, filename)
+                    print(illusory_path)
+                    if not os.path.exists(illusory_path):
+                        print(PURPLE + '- no file -\n' + ENDC)
+                    else:
+                        print(GOLD + '- Found! -\n' + ENDC)  
+                        #data = np.load(illusory_path)
+                        #print(data)
+
+
+def plot(data, column, column_order, ymax):
+    g = sns.FacetGrid(
+        data,
+        col=column,
+        col_order = column_order,
+        sharex=False,
+        size = 6,
+        aspect = .33
+    )
+
+    g.map(
+        sns.barplot,
+        "model", "time per example (seconds)", "batch",
+        hue_order = list(set(data['batch'])).sort(),
+        order = list(set(data['batch'])).sort()
+    )
+
+    if ymax == 0:
+        ymax = 1
+    else:
+        plt.yticks(np.arange(0, ymax + (ymax * .1), ymax/10))
+
+    axes = np.array(g.axes.flat)
+    #hue_start = random.random()
+    for ax in axes:
+        #ax.hlines(.0003, -0.5, 0.5, linestyle='--', linewidth=1, color=getColor(hue_start, .6, .9))
+        ax.set_ylim(0, ymax)
+
+    return plt.gcf(), axes
+
+
+def set_labels(fig, axes, labels, batch_list, model_count):
+    for i, ax in enumerate(axes):
+        increment = .75 / len(batch_list)
+        illusory = []
+        
+        if len(batch_list) % 2 == 0:        
+            foo = increment / 2
```

Comment:
need better names than foo / bar and illusory. I can't tell what exactly this code is doing